dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -8808,20 +8831,40 @@ public Contentlet checkin(Contentlet contentlet, ContentletRelationships content
 
         final Contentlet contentletReturned = checkin(contentlet, contentRelationships, cats, user, respectFrontendRoles, true, generateSystemEvent);
 
-        if (InodeUtils.isSet(contentletReturned.getInode()) && UtilMethods.isSet(selectedPermissions)) {
+        this.handlePermissions(selectedPermissions, user, respectFrontendRoles, contentletReturned);
+
+        return contentletReturned;
+    }
+
+    private void handlePermissions(final List<Permission> selectedPermissions,
+                                          final User user,
+                                          final boolean respectFrontendRoles,
+                                          final Contentlet contentlet) {
+
+        if (InodeUtils.isSet(contentlet.getInode()) && null != selectedPermissions) {
 
             final Runnable savePermissions = () -> {
 
                 try {
 
                     Logger.debug(this, ()-> "Removing the permissions for: "  + contentlet.getTitle() +
                             ", id: " + contentlet.getIdentifier());
-                    this.permissionAPI.removePermissions(contentletReturned);
+                    this.permissionAPI.removePermissions(contentlet);
+
                     Logger.debug(this, ()-> "Saving the permissions for: "  + contentlet.getTitle() +
                             ", id: " + contentlet.getIdentifier());
-                    this.permissionAPI.save(selectedPermissions.stream()
-                        .map(permission -> new Permission(contentletReturned.getPermissionId(), permission.getRoleId(), permission.getPermission()))
-                        .collect(Collectors.toList()), contentletReturned, user, respectFrontendRoles);
+
+                    selectedPermissions.stream()
+                            .map(permission -> new Permission(contentlet.getPermissionId(), permission.getRoleId(), permission.getPermission()))
+                            .forEach(permission -> {
+                                try {
+                                    Logger.debug(this, ()-> "Adding permission: " + permission);
+                                    this.permissionAPI.save(permission, contentlet, user, respectFrontendRoles);
+                                } catch (Exception e) {
+                                    Logger.error(this, e.getMessage());
```

Comment:
I would add more details on this error message, for example: contentletId, roleId and permissionId

Reply:
agree